network/manager.go

```diff
@@ -504,6 +501,12 @@ func (nm *networkManager) GetEndpointInfo(networkId string, endpointId string) (
 			epInfo.IPAddresses = append(epInfo.IPAddresses, ip.IPv6...)
 
 		}
+		if epInfo.IsEndpointStateIncomplete() {
+			epInfo, err = epInfo.GetEndpointInfoByIP(epInfo.IPAddresses, networkId)
```

Comment:
instead of overriding `epInfo` shouldn't we merge with endpoint info obtained from cns otherwise some data will be lost whatever obtained from cns

Reply:
This is the first instance that the epinfo struct is getting created. So there is no endpointInfo getting lost from CNI at this point.